paymentsheet/src/main/java/com/stripe/android/link/LinkConfigurationCoordinator.kt

```diff
@@ -103,10 +103,20 @@ internal class RealLinkConfigurationCoordinator @Inject internal constructor(
     override suspend fun attachNewCardToAccount(
         configuration: LinkConfiguration,
         paymentMethodCreateParams: PaymentMethodCreateParams
-    ): Result<LinkPaymentDetails> =
-        getLinkPaymentLauncherComponent(configuration)
-            .linkAccountManager
-            .createCardPaymentDetails(paymentMethodCreateParams)
+    ): Result<LinkPaymentDetails> {
+        val accountManager = getLinkPaymentLauncherComponent(configuration).linkAccountManager
+        val linkPaymentDetailsResult = accountManager.createCardPaymentDetails(paymentMethodCreateParams)
+        return linkPaymentDetailsResult.mapCatching { linkPaymentDetails ->
+            if (configuration.passthroughModeEnabled) {
+                accountManager.shareCardPaymentDetails(
```

Comment:
👀 I moved sharing out of `createCardPaymentDetails()` into a separate method `shareCardPaymentDetails()` and calling it separately at the callsites.